### DISCUSSION TITLE
Touchpad scrolling not smooth on Freeplane

### DISCUSSION THREAD

**Discussion by euu2021** [freeplane/freeplane#381] [General]
I'm moving from the desktop to a laptop, and I noticed that touchpad scrolling is not smooth on Freeplane. 

In this video, I first show the scroll in Chrome, and after in FP. Notice how the scroll is smooth in Chrome even with small movements. In FP, it´s more like mouse wheel scrolling.

https://user-images.githubusercontent.com/77707706/169336949-bf49e842-9bbc-4a22-acd5-9ee1e3ecb048.mp4

Can someone please confirm if it is the usual FP behavior, or if it's just my touchpad.

**Comment by euu2021** [comment]
I see that changing the settings at `Preferences…->Behaviour->Mouse wheel->Speed` changes the touchpad scroll. So, I will deduce that the touchpad, on the map, is reproducing the mouse wheel behavior.

On the other hand, the touchpad, on the menus work very well, very smooth.

Just to be more clear about the difference: 

- when moving the page in a menu window, even the smallest movement in the touchpad is recognized by FP. 
- when moving the page in the main window (the map), there is a big threshold of movement before the finger movement is recognized.

So, I will make a feature request for an improvement here.

**Comment by EdoFro** [comment]
You can smoothly move the map by left clicking on the maps background and moving the mouse. In my touch pad i get this by double tapping with one finger and then moving it on the touchpad.

This behavior was surely defined when touchpads weren't able to recognize multiple fingers on it. Maybe it's time to add two fingers gestures to  Freeplane.

bye,
edo

**Reply by euu2021** [reply]
Yes, the moving by clicking is another good example of how I would like the two fingers movement to be (in terms of being smooth). Scrolling with two fingers is more convenient, though, as the user don't need to click again after hitting corners.

> This behavior was surely defined when touchpads weren't able to recognize multiple fingers on it. Maybe it's time to add two fingers gestures to Freeplane.



Interesting. I forgot that touchpads did not recognize multiple fingers in the past.

**Comment by dpolivaev** [comment]
Unfortunately, I do not know any Java library which could process touchpad and multi-touch events. Therefore I do not think better support for touchpads can be implemented in Freeplane.

**Comment by euu2021** [comment]
Interesting. Thanks for the answer.

Now, I don't know if what I will say makes sense, but in the FP menus (for example, the preferences window), the scrolling and panning with two fingers works smoothly. So, isn't it possible to apply the same thing in the main map view?
